accepted/platform-intrinsics.md

```diff
@@ -96,30 +96,70 @@ implementations needs to be available.  As a consequence of this there can be
+         * Otherwise, the JIT throws a `BADCODE` exception. Note that this case should not occur, as the recursive IL implementation should have checked all contraints other than those the JIT is expected to expand.
+      * Otherwise (the method is not recursive), the JIT generates a regular call. This will cause the IL implementation to be invoked, and will trigger the recursive case above if it has not yet been JIT'd.
+    * Otherwise the intrinsic will be expanded in its default simple form.
+
+If the intrinsic is supported only for the constant case, and the argument passed is not a constant, then it should be treated as a regular call. This will then cause the IL implementation to be JIT'd (if it has not already been). Then we will land in the "we are recursive" case and the JIT will generate the switch statement.
+If the intrinsic is not supported for some other reason, e.g. for the given generic base type, again it should be treated as a regular call. However, in this case, the IL implementation, in addition to being recursive, should check for a valid base type and throw if it is not.
+If the JIT is unable to optimize away the intrinsic in the case where the base type doesn't match, then it should assert in Checked/Debug mode, and either generate the throw, or (simpler to implement) throw a BAD_CODE exception to the VM.
+I'm not sure if there are other preconditions to check for, but if so it would be desirable if they could fit into the same "treat it as a regular call unless it is recursive".
+
+
 ### FAQ:
 
 Q: What type and namespace should be used?  
-A: System.Runtime.CompilerServices.Intrinsics is the top namespace, but below that 
+A: System.Runtime.Intrinsics is the top namespace, but below that 
 we would expect a breakout based on architecture and platform.  Care should be 
 taken to avoid collisions - an example might be an architecture like ARM, 
 where there are multiple licensees - with the best recommendation being 
 getting feedback from the community early.
 
-Q: What should the method/parameters be named  
+Q: What should the method/parameters be named?  
 A: Simple and clear is the best bet but that doesn't always help. There
 are a lot of cases where there is prior art in C++ for these intrinsics so
 having something that parallels that implementation (maybe a bit more clearly
 named) can make the intrinsics easier to use.  That being said developers
 should try and follow regular C# naming conventions and choose names that
-indicate the semantic usage.
+indicate the semantic usage. In addition, if an equivalent intrinsic has been defined for another platform, *with the same functionality*, the naming should be consistent.
 
 Q: Is a software fallback implementation allowed? (Discussed above a bit)  
-A: Fallback is allowed but not required.  For very low level
+A: Fallback is allowed but not required, except in the immediate operand as described above.  For very low level
```

Comment:
Is there any reason a fallback should be allowed, I can't think of any cases (outside an immediate operand) where some other fallback is needed. All other cases should be able to emit the instruction (even if that is in the recursive call).

Reply:
I am with @tannergooding on not allowing software fallback except than required to support tools or reflection

Reply:
Also immediate case is a JIT/codegen fallback, not a software fallback.  The original doc was talking about a C# fallback.